### PR TITLE
Guberger master

### PR DESCRIPTION
**Pull request by blegat**
Merge changes of https://github.com/blegat/Dionysos.jl/pull/11 with master

### CODE REVIEW

**Comment by guberger** [comment]
And what about the Examples? :-)

File: src/gridspace.jl [hunk 1]
```diff
@@ -1,77 +1,92 @@
...
+    return grid_space.orig .+ pos.*grid_space.h
 end
 
-function get_pos_lim_inner(grid_space, rect)
-    lbI =  ceil.(Int, (rect.lb - grid_space.orig)./grid_space.h .+ 0.5)
-    ubI = floor.(Int, (rect.ub - grid_space.orig)./grid_space.h .- 0.5)
+function get_pos_lims_inner(grid_space, rect)
+    lbI = ceil.(Int, (rect.lb .- grid_space.orig)./grid_space.h .+ 0.5)
+    ubI = floor.(Int, (rect.ub .- grid_space.orig)./grid_space.h .- 0.5)
     return HyperRectangle(lbI, ubI)
```
[the remark this answers is not in the document]
Comment (blegat) [reply src/gridspace.jl:24]: There is no allocation being done, the HyperRectangle lb and ub are NTuples as their type automatically adapt to what is given as argument:
```julia
julia> lb = (0, 1)
(0, 1)

julia> ub = (0, 2)
(0, 2)

julia> const AB = Abstraction
Main.Abstraction

julia> isbits(AB.HyperRectangle(lb, ub))
true

julia> typeof(AB.HyperRectangle(lb, ub))
Main.Abstraction.HyperRectangle{Tuple{Int64,Int64}}
```

[the remark this answers is not in the document]
Comment (blegat) [reply src/gridspace.jl:24]: The rule of thumb is: if a structure is not mutable and only contain `isbits` fields, then it is `isbits` and creating it does not allocate.

File: src/plotting.jl [hunk 2]
```diff
@@ -140,27 +140,30 @@ end
 
 ## =============================================================================
 # Trajectory closed loop
-function trajectory_closed_loop!(ax, vars, cont_sys, trans_map, x0, nstep;
+function trajectory_closed_loop!(ax, vars, cont_sys, sym_model, x0, nstep;
```
[the remark this answers is not in the document]
Comment (blegat) [reply src/plotting.jl:143]: No, I have split the plotting in a separate submodule. The reason is that `using PyPlot` makes Travis crash so on Travis, I need the ability to not load the content of `src/plotting.jl`. This is the reason this is a separate submodule. The long-term fix might be to define recipe with https://github.com/JuliaPlots/RecipesBase.jl

File: src/plotting.jl [hunk 3]
```diff
@@ -45,8 +45,8 @@ function box!(ax, vars, lb, ub;
         fc = "green", fa = 0.5, ec = "black", ea = 1.0, ew = 1.5)
```
[the remark this answers is not in the document]
Comment (blegat) [reply src/plotting.jl:45]: That's a good idea but I leave it as future work :)

File: src/gridspace.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (guberger) [reply src/gridspace.jl:24]: Here I am not talking about allocation but probably later we will want to add some fields to HyperRectangle...

File: src/plotting.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (guberger) [reply src/plotting.jl:143]: Ok :-)

File: src/gridspace.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (blegat) [reply src/gridspace.jl:24]: I don't think we should add any more field. I you wand more info, you can have a structure with `HyperRectangle` as a subfield and this one have more fields.

[the remark this answers is not in the document]
Comment (guberger) [reply src/gridspace.jl:24]: Ok! fine :-)

**Pull request merged**